### PR TITLE
fix: allowed output file protocols on Windows

### PR DESCRIPTION
**Pull request by alexpozzi**
Fixes https://github.com/KnpLabs/snappy/issues/489

### CODE REVIEW

**Review by AntoineLelaisant — DISMISSED** [review]

File: src/Knp/Snappy/AbstractGenerator.php [hunk 1]
```diff
@@ -630,8 +630,16 @@ protected function prepareOutput($filename, $overwrite)
...
         $scheme = isset($parsedFilename['scheme']) ? \mb_strtolower($parsedFilename['scheme']) : '';
-        if ($scheme !== '' && $scheme !== 'file') {
-            throw new InvalidArgumentException(\sprintf('The output file scheme is not supported. Expected \'\' or \'file\' but got \'%s\'.', $scheme));
+        if (!(
+            $scheme === '' ||
+            $scheme === 'file' ||
+            (
+                // Check if it's a Windows path
+                \strlen($scheme) === 1 &&
+                \preg_match('/^[a-z]:(?:[\\\\\/]?(?:[\w\s!#()-]+|[\.]{1,2})+)*[\\\\\/]?/i', $filename) === 1
```
Comment (PedroTroller) [inline src/Knp/Snappy/AbstractGenerator.php:639]: Can you give some exemples with specs for example ? Maybe set the regex in a constant will clarify the condition.

Comment (PedroTroller) [inline src/Knp/Snappy/AbstractGenerator.php:639]: `if (!(a || b || (c && d))` is not realy clear also :wink:

**Review by PedroTroller — CHANGES_REQUESTED** [review]

**Review by AntoineLelaisant — APPROVED** [review]

File: src/Knp/Snappy/AbstractGenerator.php [hunk 2]
```diff
@@ -21,6 +21,10 @@ abstract class AbstractGenerator implements GeneratorInterface, LoggerAwareInter
 {
     use LoggerAwareTrait;
 
+    protected const ALLOWED_PROTOCOLS = ['file'];
+
+    protected const WINDOWS_LOCAL_FILENAME_REGEX = '/^[a-z]:(?:[\\\\\/]?(?:[\w\s!#()-]+|[\.]{1,2})+)*[\\\\\/]?/i';
```
Comment (PedroTroller) [inline src/Knp/Snappy/AbstractGenerator.php:26]: Is the first letter always lower case? C:` won't happen?

Comment (alexpozzi) [reply src/Knp/Snappy/AbstractGenerator.php:26]: No, but the regular expression is case-insensitive, so no need of adding `A-Z`.

Comment (PedroTroller) [reply src/Knp/Snappy/AbstractGenerator.php:26]: `/i`.... indeed.

**Review by PedroTroller — APPROVED** [review]

**Pull request merged**